internal/planning/deploy/deploy.go

```diff
@@ -782,12 +779,10 @@ func prepareRunOpts(ctx context.Context, stack *planning.Stack, srv planning.Ser
 		return err
 	}
 
-	if merged.WorkingDir != "" {
-		out.MainContainer.WorkingDir = merged.WorkingDir
+	err = out.MainContainer.BinaryConfig.MergeFrom(merged)
+	if err != nil {
+		return err
 	}
-	out.MainContainer.Args = append(out.MainContainer.Args, merged.Args...)
-	out.MainContainer.Env = append(out.MainContainer.Env, srv.Proto().MainContainer.Env...)
-	out.MainContainer.Env = append(out.MainContainer.Env, merged.Env...)
```

Comment:
Where do these merges occur now? In the call above? Does that mean that there are more fields being changed, not just Args and Env?